cross-validation/logistic-regression/script.whizzml

```diff
@@ -0,0 +1,404 @@
+;;   objective-name: (string) Objective field name
+;;   dataset-name: (string) Name of the origin dataset
+;;   model-options: (map) Options used to build the models/ensembles
+;;   evaluation-options: (map) Options used to build evaluations
+;;
+;; Output: (list) List of evaluation IDs
+;;
+(define (create-k-evaluations dataset-ids
+                              objective-name
+                              dataset-name
+                              model-options
+                              evaluation-options)
+  (let (number-of-models (get model-options "number_of_models" 1)
+        regularization (get model-options "regularization" false)
+        k-fold-pairs (pair-k-folds dataset-ids)
+        ensemble-options (select-keys model-options ENSEMBLE_OPTIONS)
+        model-options (select-keys model-options MODEL_OPTIONS)
+        logistic-options (select-keys model-options LOGISTIC_OPTIONS)
+        evaluation-options (select-keys evaluation-options EVALUATION_OPTIONS)
+        type (if (> number-of-models 1) "ensemble" "model")
+        multidatasets (map (lambda (x) (nth x 1)) k-fold-pairs)
+        models (cond (> number-of-models 1)
+                     (create-k-models type
+                                      multidatasets
+                                      objective-name
+                                      ensemble-options)
+                     regularization
+                     (create-k-models type
+                                      multidatasets
+                                      objective-name
+                                      logistic-options)
+                     (create-k-models type
+                                      multidatasets
+                                      objective-name
+                                      model-options))
+        evaluations (loop (index 0 evaluations [])
+                      (if (= index (count dataset-ids))
+                          evaluations
+                          (let (name (str (+ index 1)
+                                          "-fold Evaluation "
+                                          dataset-name)
+                                evaluations
+                                (append evaluations
+                                        (create-evaluation
+                                         (merge (assoc {}
+                                                       type
+                                                       (nth models index)
+                                                       "dataset"
+                                                       (nth dataset-ids
+                                                            index)
+                                                       "name"
+                                                       name)
+                                                evaluation-options))))
+                          (recur (+ index 1) evaluations)))))
```

Comment:
this can be simplified with `iterate` and also with the new ability of
create evaluation of taking directly dataset and model (of any type)
ids:


```
evaluations (iterate (es []
                      id dataset-ids
                      mid models
                      idx (range 1 (+ 1 (count-dataset-ids))))
              (let (name (str idx "-fold evaluation " dataset-name)
                    opts (assoc evaluation-options "name" name))
                (append es (create-evaluation id mid opts))))
```

(or you can just use `idx` and then `(dataset-ids (- idx 1))` `(models
(- idx 1))` in the body)